app/api/__init__.py

```diff
@@ -141,91 +158,92 @@ def get_token(id):
-                message_image=data.get("message_image", False),
-                message_command=data.get("message_command", False),
-                message_broadcast=data.get("message_broadcast", False),
-                room_query=data.get("room_query", False),
-                room_log_query=data.get("room_log_query", False),
-                room_create=data.get("room_create", False),
-                room_update=data.get("room_update", False),
-                room_delete=data.get("room_delete", False),
-                layout_query=data.get("layout_query", False),
-                layout_create=data.get("layout_create", False),
-                layout_update=data.get("layout_update", False),
-                task_create=data.get("task_create", False),
-                task_query=data.get("task_query", False),
-                task_update=data.get("task_update", False),
-                token_generate=data.get("token_generate", False),
-                token_query=data.get("token_query", False),
-                token_update=data.get("token_update", False),
-                token_invalidate=data.get("token_invalidate", False),
-            )
+    token = Token(
+        task=task,
+        room=room,
+        source=data.get("source", None),
+        permissions=Permissions(
+            user_query=data.get("user_query", False),
+            user_log_event=data.get("user_log_event", False),
+            user_room_join=data.get("user_room_join", False),
+            user_room_leave=data.get("user_room_leave", False),
+            message_text=data.get("message_text", False),
+            message_image=data.get("message_image", False),
+            message_command=data.get("message_command", False),
+            message_broadcast=data.get("message_broadcast", False),
+            room_query=data.get("room_query", False),
+            room_log_query=data.get("room_log_query", False),
+            room_create=data.get("room_create", False),
+            room_update=data.get("room_update", False),
+            room_delete=data.get("room_delete", False),
+            layout_query=data.get("layout_query", False),
+            layout_create=data.get("layout_create", False),
+            layout_update=data.get("layout_update", False),
+            task_create=data.get("task_create", False),
+            task_query=data.get("task_query", False),
+            task_update=data.get("task_update", False),
+            token_generate=data.get("token_generate", False),
+            token_query=data.get("token_query", False),
+            token_update=data.get("token_update", False),
+            token_invalidate=data.get("token_invalidate", False),
         )
-        db.session.add(token)
-        db.session.commit()
-        return jsonify(token.id)
-    except (IntegrityError, StatementError) as e:
-        return make_response(jsonify({'error': str(e)}), 400)
+    )
+    db.add(token)
+    db.commit()
+
+    return make_response(token.as_dict())
```

Comment:
Returning `token.id` only totally makes sense as the user has just created the token with the specified values. The `id` is the only value they don't know. If the token shall be verified, the corresponding `GET` method should be used in addition. However, returning the plain `id` (which would be returned as a string), can't be parsed by `jq`. I'd stick with returning a dictionary/JSON-string, but only with the `id` as key. In any case, this should be unified over all `POST` and `PUT` methods. Do you favor one over the other?

Reply:
No real preference at the moment on my side. I can understand if one would want to return the whole dictionary, in this way you (or any bot) get a single object with passed and new information that you could maybe pass to other functions or return and don't have to create one yourself. And in my understanding that's also the approach that all other POST and PUT methods use?  So less things to change.
But if you want to unify this over all POST and PUT requests and we follow the philosophy 'return only what you did not know before' ... what do you have in mind for the Rooms POST request? In my understanding the equivalent to 'id' here would be 'name' for rooms (our primary key), but the 'name' is something we pass to a POST request. While checking that I noticed that the Rooms table is the only one that does not inherit from Common. I assume there exists a reason for that? Then it would be a special case anyway.

Reply:
> what do you have in mind for the Rooms POST request?

Uhm... Yes! 😆
Making `Room.name` a key was required, but using it as a primary key wasn't a good choice for several reasons. `Common` provides `id` as primary key so `Room` could not inherit from `Common`. (We didn't use Sqlalchemy at first). So, yes, this is a special case we might want to unify.
Cc @janagoetze

Reply:
Just to opt-in, I think the way to go is to give rooms an `id`. The user of the api can then requests "all" rooms and have a look through them (to potentially get the names). 

We could optionally also provide a `query` endpoint for rooms, but when the name really matters, then there are probably not many rooms (manual setup) and when there are many rooms, then the name doesnt matter so much, because rooms are then handled automatically (using the ids).

Reply:
Just for reference: Currently, I'm fighting with the API of OpenVidu. The REST API there always returns the full struct for creating, getting, and modifying data. Even when getting for example a list of all active sessions, it returns the full struct for every entry and is laid out like this:
```JSON
{
    "numberOfElements": 0,
    "content": []
}
```